lib/pbench/server/auth/auth.py

```diff
@@ -7,19 +8,65 @@
 from flask_httpauth import HTTPTokenAuth
 import jwt
 
+from pbench.server import PbenchServerConfig
 from pbench.server.auth import OpenIDClient, OpenIDClientError
 from pbench.server.database.models.active_tokens import ActiveTokens
 from pbench.server.database.models.users import User
 
 
+@dataclass
+class InternalUser:
+    """Internal user class for storing user related fields fetched
+    from OIDC token decode.
+    """
+
+    id: str
+    username: str
+    email: str
+    first_name: Optional[str] = None
+    last_name: Optional[str] = None
+    roles: Optional[list[str]] = None
+
+    def __str__(self) -> str:
+        return f"User, id: {self.id}, username: {self.username}"
+
+    def is_admin(self):
+        return "ADMIN" in self.roles
+
+
 class Auth:
     token_auth = HTTPTokenAuth("Bearer")
+    oidc_client: OpenIDClient = None
 
     @staticmethod
```

Comment:
I need some help.  As far as I can see, this is a really weird class:  all of its methods are static, except for two, and the second of those doesn't reference `self` (and therefore could/should be static) and the first only uses `self` to call the second (so it could/should be static too!).

Apparently "static classes" are not a Python (or Pythonic) Thing -- were supposed to just use module-level functions.  But this class is tied up with some sort of [automatic token verification](https://github.com/distributed-system-analysis/pbench/blob/ee0c18fac7ad318120e9575a8a01e26e26b33e79/lib/pbench/server/auth/auth.py#L134) which I don't recall...so I don't really know what to recommend.

But, it seems like we don't have things set up quite the way we should here, which is what is leading to this weird discussion about how to set `oidc_client`....